src/bots/telegram.ts

```diff
@@ -305,50 +310,76 @@ export default class TelegramBot extends BotClient {
   }
 
   public async start(): Promise<boolean> {
-    try {
-      if (this.token) {
-        await this.bot.startPolling({ restart: true });
-        this.isRunning = true;
-        // Handle being removed from chats (except channels apparently)
-        this.bot.on('left_chat_member', async (message) => {
-          const leftMember = message.left_chat_member;
-          const telegramUser = await this.bot.getMe();
-          const userID = telegramUser.id;
-
-          if (!leftMember || leftMember.id !== userID) {
-            // It's not the bot
-            return;
-          }
+    if (!this.enabled) {
+      this.logger.info('Autostart disabled.');
+      return false;
+    }
+    assertIsDefined(this.token, `Token is undefined`);
```

Comment:
How excactly does the bot behave when this is undefined?
Not that this will happen every time that a new contributor starts the bot for the first time.

I think I liked that it gave a warning and didn't start the bot, but didn't crash anything

Reply:
It throws an error that the main file catches. I'm fine with wrapping this in a huge try-catch but there are some nested `try-catch`s too there and it kind of looked like a mess to me